drivers/kscan/kscan_matrix_handlers.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2022 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/kscan.h>
+#include <syscall_handler.h>
+
+static int z_vrfy_kscan_matrix_configure(const struct device *dev,
+					 kscan_isr_callback_t isr_callback)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_KSCAN(dev, matrix_config));
+
+	return z_impl_kscan_matrix_configure((const struct device *)dev, isr_callback);
```

Comment:
This similarly looks like a security vulnerability to me. The `isr_callback` is a function pointer that is attacker-controlled, and is saved by the implementation to be invoked from (privileged mode) ISR context by the driver here:

```
	if (data->isr_callback != NULL) {
		data->isr_callback(dev);
	}
```

This similarly seems to allow the thread to escape its sandbox.

Reply:
ISR callback registration for drivers should never be done from user mode.

Reply:
Thanks for confirming both issues @dcp

Reply:
I removed the callback function register API from the kscan_matrix_handler.